packages/@lwc/wds-core/src/node/plugins/mock-esm/handle/mock-resolved.js

```diff
@@ -62,6 +69,20 @@ ${withoutDefault(exportedNames)
       __liveDefault__ = newExports.default;
     }
   },
+  async update(exportName,code){
+      if (exportName === 'default') {
+        const updatedCode = code.replace(/\{exportName\}/g, '__liveDefault__');
+        const executeCode = new Function( '__liveDefault__',updatedCode);
+        const obj = exportObjects.get(exportName); 
+        executeCode(obj);
+      }
+      else {
+        const updatedCode = code.replace(/\{exportName\}/g, exportName);
+        const executeCode = new Function(exportName,updatedCode);
+        const obj = exportObjects.get(exportName); 
+        executeCode(obj);
+      }   
+  },
```

Comment:
Instead of this `update` function, I think we'll want to expose an `eval` function that allows for interacting with exported things. It might be implemented like so:

```javascript
`
  async eval(codeToEval) {
    const AsyncFunction = async function () {}.constructor;
    const fn = new AsyncFunction(
      ${[...withoutDefault(exportedNames).map(name => `'${name}'`), 'codeToEval'].join(', ')}
    );
    return fn(${withoutDefault(exportedNames).join(', `')});
  }
`
```

- accept some code to eval
- construct a new async function
   - it should take N parameters, where each parameter corresponds to a named export
   - it should use the provided code for its function body
- invoke that function, passing in the named exports as arguments, in the same order as they were declared as params
- return the `Promise`